packages/@headlessui-react/src/components/transition/transition.tsx

```diff
@@ -463,6 +463,7 @@ function TransitionChildFn<TTag extends ElementType = typeof DEFAULT_TRANSITION_
         // Map data attributes to `entered` class (backwards compatibility)
         !slot.transition && show && entered
       )?.trim() || undefined, // If `className` is an empty string, we can omit it
+    ...transitionDataAttributes(slot),
```

Comment:
We should probably rename `slot` to `transitionData` just so we don't accidentally add this back